packages/debian/metalk8s-sosreport-deb-2.4.0/debian/src/containerd.py

```diff
@@ -0,0 +1,73 @@
+#! /bin/env python
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+
+
+class containerd(Plugin, RedHatPlugin, UbuntuPlugin):
```

Comment:
If we do keep two versions of the code, then I think the Ubuntu version doesn't need to inherit from `RedHatPlugin` and the CentOS version doesn't need to inherit from `UbuntuPlugin`.

But seeing the quantity of duplicated code, I think we will want to keep a single implementation that support both OSes.

Worst case (if they do diverge a lot) we could have something like:

```python
class UbuntuContainerd(Plugin, UbuntuPlugin):
    …

class RedHatContainerd(Plugin, RedHatPlugin):
    …

if os_is_ubuntu:
     containerd = UbuntuContainerd
elif os_is_redhat:
     containerd = RedHatContainerd
else:
    raise some_error
```

but that won't be needed for now.

Reply:
In the meantime, this is a possibility for inheriting from the correct plugin:

```
In [1]: class Both: 
   ...:     pass 
   ...:                                                             

In [2]: class One: 
   ...:      
   ...:     def a(self): 
   ...:         print('toto') 
   ...:                                                             

In [3]: class Other: 
   ...:     def a(self): 
   ...:         print('tata') 
   ...:                                                             

In [4]: selector = True                                             

In [5]: class Inheritor(Both, One if selector else Other): 
   ...:     pass 
   ...:                                                             

In [6]: i = Inheritor()                                             

In [7]: i.a()                                                       
toto
```